cdap-tms/src/main/java/co/cask/cdap/messaging/store/AbstractMessageTable.java

```diff
@@ -117,13 +117,26 @@ public void delete(TopicId topicId, long startTimestamp, short startSequenceId,
     delete(startRow, Bytes.stopKeyForPrefix(stopRow));
   }
 
+  protected long decodeTx(byte[] txPtr) {
+    return Bytes.toLong(txPtr);
+  }
+
+  protected byte[] encodeTx(long txPtr) {
```

Comment:
It can be easily done with this:

```java
    int inc = 1;
    for (int i = bytes.length - 1; i >= 0; i--) {
      int val = (bytes[i] ^ 0xff) + inc;
      bytes[i] = (byte) val;
      if (val <= 0xff) {
        inc = 0;
      }
    }
```